src/org/labkey/test/tests/UserClonePermissionTest.java

```diff
@@ -0,0 +1,179 @@
+        _permissionsHelper = new ApiPermissionsHelper(getProjectName());
+        _permissionsHelper.assertPermissionSetting(CLONED_TARGET_SITE_USER, "Author");
+        _permissionsHelper.assertPermissionSetting(CLONED_TARGET_SITE_USER, "Editor");
+        _permissionsHelper.assertNoPermission(CLONED_TARGET_SITE_USER, "Submitter");
+
+        log("Verifying site level permission");
+        _permissionsHelper = new ApiPermissionsHelper("");
+        _permissionsHelper.assertPermissionSetting(CLONED_TARGET_SITE_USER, "PlatformDeveloper");
+        _permissionsHelper.assertPermissionSetting(CLONED_TARGET_SITE_USER, "SiteAdmin");
+
+        log("Verify audit log entry");
+        DataRegionTable table = goToAdminConsole().clickAuditLog()
+                .selectView("Group and role events")
+                .getLogTable();
+        table.setFilter("user", "Contains", TARGET_SITE_USERNAME);
+
+        Assert.assertEquals("Incorrect audit rows for " + TARGET_SITE_USERNAME, 7, table.getDataRowCount());
+        Assert.assertEquals("Incorrect audit log messages for " + TARGET_SITE_USERNAME, Arrays.asList("The user " + CLONED_TARGET_SITE_USER + " was assigned to the security role Author.",
+                        "User: " + CLONED_TARGET_SITE_USER + " was added as a member to Group: Developers",
+                        "User: " + CLONED_TARGET_SITE_USER + " was added as a member to Group: Test clone group",
+                        "User: " + CLONED_TARGET_SITE_USER + " was added as a member to Group: Administrators",
+                        "The user " + CLONED_TARGET_SITE_USER + " was removed from the security role Submitter.",
+                        "The user " + CLONED_TARGET_SITE_USER + " had their group memberships and role assignments deleted and replaced with those of user sourcesiteuser@clonepermission.test",
+                        "The user " + CLONED_TARGET_SITE_USER + " was assigned to the security role Submitter."),
+                table.getColumnDataAsText("comment"));
+    }
+
+    @Test
+    public void testAppAdminClonePermission()
+    {
+        goToSiteUsers();
+        impersonateRole("Application Admin");
+        clickAndWait(Locator.linkWithText(TARGET_APP_USERNAME));
+        clickAndWait(Locator.linkWithText("Clone Permissions"));
+        ClonePermissionsPage clonePermissionsPage = new ClonePermissionsPage(getDriver());
+
+        Assert.assertEquals("Incorrect warning message",
+                "Warning! Cloning permissions will delete all group memberships and direct role assignments for " + TARGET_APP_USERNAME + " (" +
+                        CLONED_TARGET_APP_USER + ") and replace them with the group memberships and direct role assignments of the user selected below.",
+                clonePermissionsPage.getWarningMessage());
+
+        clonePermissionsPage.setCloneUser(CLONED_SOURCE_APP_USER);
+        clonePermissionsPage.clonePermission();
+        stopImpersonating();
+
```

Comment:
This should verify that an app admin cannot clone permissions from a site admin.